### PR TITLE
Add rate limit checking

### PR DESCRIPTION
**Pull request by chartgerink**
This PR adds a small function to check the rate limit on the GitHub API. When this hits 0, it will display a warning.

This will help people get feedback on when they are exceeding the rate limit, which can be helpful to scope down the task to be more manageable. 

Pre-discussed in #36. Happy to take feedback as you know your standards for this package best. I tried to stick to how you've previously implemented warnings for sake of package consistency.

### CODE REVIEW

File: R/github.R [hunk 1]
```diff
@@ -537,3 +540,20 @@ get_gh_contrib_issue <- function (org, repo) {
 
     unlist (pings)
 }
+
+#' Check the GitHub rate limit and warn if exceeded.
+#'
+#' @noRd
+check_rate_limit <- function () {
+    gh_state <- gh::gh_rate_limit()
+    if (gh_state$remaining == 0) {
```
Comment (mpadge) [inline R/github.R:549]: Just a suggestion, but how about an additional check like this:
``` r
limit_warn <- 0.1
warn_txt <- NULL
if (gh_state$remaining / gh_state$limit < limit_warn) {
  warn_txt <- "You have used > 90% of your GitHub calls..."
} else if (gh_state$remaining == 0) {
  warn_txt <- "The GitHub rate limit has been reached ..."
}
if (!is.null (warn_txt)) {
  cli::cli_alert_warning (warn_txt, wrap = FALSE)
}
```
GitHub itself now warns on several things at a limit of 75%, but I think for rate calls which reset every hour that 90% should be fine. Oh, and we should definitely also mention that it is an *hourly* limit.

**Comment by mpadge** [comment]
@chartgerink That looks great! Could you please fix the tests? They include mock results of all external calls, so now fail because new API calls have been introduced. You should be able to update them with these steps:

1. Delete the `tests/testthat/getcontribs` directory (not `git rm`, just `rm`!)
2. Run
    ``` r
    library (httptest2)
    testthat::test_local ()
    ```
3. `git add tests/testthat/getcontribs`

Then you can stage all of those files, commit, and push. Thanks!

**Comment by chartgerink** [comment]
Thanks @mpadge - updated accordingly. 👍

**Comment by mpadge** [comment]
Sorry for dropping this @chartgerink. This still fails tests, but I'll merge and fix everything straight away, before moving straight on to #35.

**Pull request merged**